Day 5. Homework

eclimov opened the pull request:
+ Controller upgrades
+ Template upgrades
+ Entity upgrades
+ Repository

eclimov committed:
- Now showing only active not expired jobs on main page
- "showAction" method added in "Job" controller. Its purpose is to render template with description of a job, provided in arguments (id)
- Updated "JobFixture":
now "expiresAt" datetime is calculated differently, increasing chances of both expired and not expired jobs available in DB after "fixtures:load" command,
"position" word is more "real" now
- now creating "createdAt" and "expiresAt" dates, using lifecycle callbacks rather than classes' constructors
- Created "jobRepository"
- Each job position on the main page is a link to its own page, containing the detailed description
- Updated main page's template: it is more user-friendly now

gregurco commented on src/DataFixtures/JobFixtures.php:
```diff
@@ -25,15 +25,15 @@ public function load(ObjectManager $manager): void
             $job->setDescription($faker->paragraph);
             $job->setEmail($faker->email);
-            $job->setExpiresAt($faker->dateTime);
+            $job->setExpiresAt($faker->dateTimeBetween($startDate = '-20 days', $endDate = '+20 days'));
```
no need to write in this way and to introduce 2 more variables, just write:
```
$job->setExpiresAt($faker->dateTimeBetween('-20 days', '+20 days'));
```

gregurco commented on src/Repository/JobRepository.php:
```diff
@@ -0,0 +1,26 @@
+use App\Entity\Job;
+use \Doctrine\ORM\EntityRepository;
+
+class JobRepository extends EntityRepository
```
you created repository in SF3 way. See how it should be in SF4:
https://symfony.com/doc/current/doctrine.html#querying-for-objects-the-repository

gregurco commented on src/Repository/JobRepository.php:
```diff
@@ -0,0 +1,26 @@
+            ->andWhere('j.activated = :activated')
+            ->setParameters([
+                'datetime' => new \DateTime(),
+                'activated' => true
```
add `,` at the last element in array too

eclimov committed:
- Removed unnecessary variables' declarations in "JobFixtures"
- Updated "JobRepository": now it is created the Symfony4 way
- Added trailing comma to array, according to coding style rules

eclimov commented:
Fixed
